rekallpy/rekall/interval_set.py

```diff
@@ -809,32 +810,63 @@ def coalesce(self,
         Returns:
             A new IntervalSet of intervals that are disjoint along ``axis`` and
             are at least ``epsilon`` apart.
+            
         """
-
-        def update_output(output, intrvl):
-            if len(output) == 0:
-                output.append(intrvl)
-            else:
-                merge_candidate = output[-1]
+        if (len(self._intrvls) == 0):
+            return self
+
+        new_coalesced_intrvls = []
+        current_intrvls = []
```

Comment:
Can you add a comment describing what current_intrvls represents throughout the algorithm?